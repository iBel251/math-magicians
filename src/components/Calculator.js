import React from 'react';

const keys = [
  {
    name: 'AC',
    class: 'btn gray',
  },
  {
    name: '+/-',
    class: 'btn gray',
  },
  {
    name: '%',
    class: 'btn gray',
  },
  {
    name: 'รท',
    class: 'btn orange',
  },
  {
    name: '7',
    class: 'btn gray',
  },
  {
    name: '8',
    class: 'btn gray',
  },
  {
    name: '9',
    class: 'btn gray',
  },
  {
    name: 'x',
    class: 'btn orange',
  },
  {
    name: '4',
    class: 'btn gray',
  },
  {
    name: '5',
    class: 'btn gray',
  },
  {
    name: '6',
    class: 'btn gray',
  },
  {
    name: '-',
    class: 'btn orange',
  },
  {
    name: '1',
    class: 'btn gray',
  },
  {
    name: '2',
    class: 'btn gray',
  },
  {
    name: '3',
    class: 'btn gray',
  },
  {
    name: '+',
    class: 'btn orange',
  },
  {
    name: '0',
    class: 'btn gray zero',
  },
  {
    name: '.',
    class: 'btn gray',
  },
  {
    name: '=',
    class: 'btn orange',
  },
];
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="container">
        <from>
          <input type="text" placeholder="0" />
        </from>
        <div className="keypad">
          {keys.map((btn) => (<button type="button" key={btn.name} className={btn.class}>{btn.name}</button>))}
        </div>
      </div>
    );
  }
}

export default Calculator;
