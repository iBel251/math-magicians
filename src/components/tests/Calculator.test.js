import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

test('renders the quote correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

// describe('Calculator properly functions', () => {
//     test('returns 120 after summing 56 and 64', () => {
//         render(<Calculator />)
//         fireEvent.click(screen.getByRole('button', {name: '5'}))
//         fireEvent.click(screen.getByRole('button', {name: '6'}))
//         fireEvent.click(screen.getByRole('button', {name: '+'}))
//         fireEvent.click(screen.getByRole('button', {name: '6'}))
//         fireEvent.click(screen.getByRole('button', {name: '4'}))
//         fireEvent.click(screen.getByRole('button', {name: '='}))
        
//         const result = waitFor(() => screen.getByText('120'));
//         expect(result.textContent).toBe('120');
//     })
// })