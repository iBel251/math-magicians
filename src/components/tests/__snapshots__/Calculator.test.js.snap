// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the quote correctly 1`] = `
<div
  className="window"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="container"
  >
    <div
      className="result"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="btn orange"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="btn orange"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="btn orange"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="btn orange"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="btn gray zero"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="btn gray"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="btn orange"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
